pkg/daemon/update.go

```diff
@@ -2738,13 +2799,75 @@ func (dn *CoreOSDaemon) applyLayeredOSChanges(mcDiff machineConfigDiff, oldConfi
 	var osExtensionsContentDir string
 	var err error
 
+	// For image mode status reporting we need the node's MCP association to populate it's MCN
+	imageModeStatusReportingEnabled := dn.fgHandler != nil && dn.fgHandler.Enabled(features.FeatureGateImageModeStatusReporting)
+	pool := ""
+	if imageModeStatusReportingEnabled {
+		pool, err = helpers.GetPrimaryPoolNameForMCN(dn.mcpLister, dn.node)
+		if err != nil {
+			return err
+		}
+	}
+
 	if newConfig.Spec.BaseOSExtensionsContainerImage != "" && (mcDiff.osUpdate || mcDiff.extensions || mcDiff.kernelType) && !mcDiff.oclEnabled {
 		// TODO(jkyros): the original intent was that we use the extensions container as a service, but that currently results
 		// in a lot of complexity due to boostrap and firstboot where the service isn't easily available, so for now we are going
 		// to extract them to disk like we did previously.
+
+		// Report ImagePulledFromRegistry condition as unknown (pulling)
+		if imageModeStatusReportingEnabled {
+			err := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+				&upgrademonitor.Condition{State: mcfgv1.MachineConfigNodeImagePulledFromRegistry, Reason: string(mcfgv1.MachineConfigNodeImagePulledFromRegistry), Message: fmt.Sprintf("Pulling image %s from registry", newConfig.Spec.BaseOSExtensionsContainerImage)},
```

Comment:
Like how we have `pulling OS Image` as the message above, does it make sense to specify that this is `Pulling extensions image` or something like that?